pkg/controllers/user/alert/watcher/node.go

```diff
@@ -0,0 +1,172 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/controllers/user/alert/manager"
+	"github.com/rancher/rancher/pkg/ticker"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+type NodeWatcher struct {
+	machineLister      v3.NodeLister
```

Comment:
is there any specific reason why we need machineLister here? Machine is just a representation of the node (nodesyncer.go does the propagation).

Reply:
Ok, looks like we are using nodesyncer generated info (Requests/Limits)